contracts/delegatecall/LibEtherForENS.sol

```diff
@@ -0,0 +1,25 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+pragma solidity ^0.8.20;
+
+abstract contract Resolver {
+    function addr(bytes32 node) public view virtual returns (address);
+}
+
+abstract contract ENS {
+    function resolver(bytes32 node) public view virtual returns (Resolver);
+}
+
+library LibEtherForENS {
+    function etherForENS(bytes32 node, uint256 value, bytes memory payload) external{
+        // address of The ENS registry
+        ENS ens = ENS(0x00000000000C2E074eC69A0dFb2997BA6C7d2e1e);
+
+        Resolver resolver = ens.resolver(node);
+        address resolution = resolver.addr(node);
```

Comment:
Why not `receiver` instead of `resolution`?

Reply:
Yes we can. It's a different perspective. `recipient` is viewed from the transfer, `resolution` is viewed from the ENS. I will rename it to `recipient`.